scripts/compactor_runner.py

```diff
@@ -62,13 +62,18 @@ class CommandBuilder(object):
     def __init__(self, config):
         self._config = config
 
-    def derive_xmx_value(self, diskBacked):
+    def derive_xmx_value(self, diskBacked, compactor_config):
         try:
             mem_bytes = os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES')
             mem_mb = int(mem_bytes / (1024. ** 2))
             xmx = max(512, int(mem_mb * 0.04))
             if diskBacked is True:
-                xmx = max(self._config.xmx_min_in_mb, int(mem_mb * float(self._config.xmx_perc)/100))
+                if xmx > 1000 and "Large" in compactor_config["MemoryOptions"]:
+                    xmx = compactor_config["MemoryOptions"]["Large"]
+                elif xmx > 512 and xmx < 1000 and "Medium" in compactor_config["MemoryOptions"]:
+                    xmx = compactor_config["MemoryOptions"]["Medium"]
+                else:
+                    xmx = max(self._config.xmx_min_in_mb, int(mem_mb * float(self._config.xmx_perc)/100))
```

Comment:
Change it to `compactor_config["MemoryOptions"]["Small"]`?

Reply:
Added a line for small and retained the else part in case there's no Xmx value passed in the config file